tests/app/polyglot-opentracing/java/src/main/java/org/hawkular/apm/tests/app/polyglot/swarm/rest/HelloHandler.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.tests.app.polyglot.swarm.rest;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+import org.hawkular.apm.tests.app.polyglot.swarm.cdi.APMTracer;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMapExtractAdapter;
+
+/**
+ * @author Pavol Loffay
+ */
+@Path("/")
+public class HelloHandler {
+
+    @Inject
+    @APMTracer
```

Comment:
I disagree here, tracer implementations does not have to provide CDI integration, therefore it's better to use a producer, however here has to be a qualifier because our impl is packaged with `beans.xml` and there would ambiguous dependencies. 

I would prefer if we did not include `beans.xml` in our provider. If we want to then create an integration jar.